_posts/2016-05-12-JS-Conf-Budapest-day-1.md

```diff
@@ -0,0 +1,775 @@
+* [Maurice de Beijer: Event-sourcing your React-Redux applications](#maurice-de-beijer-event-sourcing-your-react-redux-applications)
+* [Rachel Watson: The Internet of Cats](#rachel-watson-the-internet-of-cats)
+* [Nick Hehr: The other side of empathy](#nick-hehr-the-other-side-of-empathy)
+
+
+****
+
+
+## Day 1: Morning
+
+<img class="p-image float-image" width="200" alt="Suz Hinton" src="/img/js-conf-budapest/speaker-seldo.jpg">
+
+### Laurie Voss: What everybody should know about npm
+
+Laurie is CTO at npm Inc.
+
+You can find him on Twitter using the handle [@seldo](https://twitter.com/seldo).
+
+The presentation he gave can be found a [slides.com/seldo/jsconf-budapest](http://slides.com/seldo/jsconf-budapest).
+
+<blockquote class="clear"><p>
+npm is six years old, but 80% of npm users turned up in the last year.
+That's a lot of new people! Because of that, a lot of older, core features aren't known about by the majority of npm users.
+This talk is about how npm expects you to use npm, and the commands and workflows that can make you into a power user.
+There will be lots of stuff for beginners, and definitely some tricks that even most pros don't know.
+</p></blockquote>
+
+#### How does npm look up packages?
+In contrast to what most people think, npm does not download it's modules from GitHub or other version control systems.
+They would not like it that such an amount of data is transferred on a daily basis.
+
+In short npm does this: You -> CLI -> Registry.
+Let's dive in.
+
+1. First npm will take a look at your local cache and see if the package your are looking for is present.
+2. Next it will resort to the CDN network and will use a server whos is the closest to your position as possible.
+3. Finally, if npm can't find the package in local cache or the CDN network, it will look up in the registry. The registry is a set of servers all around the world and will try to match the best version that you are looking for.
+
+#### EACCESS error
+A lot of people have issues with EACCESS errors because they used sudo to install things.
+The easy solution is to always keep on using sudo BUT we can [easily fix npm permission issues](https://docs.npmjs.com/getting-started/fixing-npm-permissions).
+
+#### package.json
+Don't write your `package.json` yourself, let NPM do it!
+It will always do it better. Use `npm init` which will ask some basic questions and will generate `package.json` for you.
+
+#### Scopes
+A new feature in npm is scopes.
+These are modules that are "scoped" under an organization name that begins with `@`.
+Scopes can be public and private.
+Here is how to use scopes:
+
+{% highlight sh %}
+npm init --scope=username
+npm install @myusername/mypackage
+require('@myusername/mypackage')
+{% endhighlight %}
+
+#### npm-init.js
+To extend the `npm init` command it is possible to create an `npm-init.js` file.
```

Comment:
**It is possible** to extend the `npm init` command ~~it is possible to create~~  **by creating** an `npm-init.js` file.